src/osd/ECBackend.cc

```diff
@@ -500,6 +500,8 @@ void ECBackend::continue_recovery_op(
 	assert(!op.recovery_progress.first);
 	dout(10) << __func__ << ": canceling recovery op for obj " << op.hoid
 		 << dendl;
+	// XXX: Doing this on the read error handling just loops in recovery
+	// How would this case ever actually happen?
```

Comment:
Yep, that's what I was talking about.  Like I said, here, it's ok because ReplicatedPG/PG already knows that the source died and won't retry until the object is found again.  You need to call something which updates missing_loc (which I think you do earlier in this series with _failed_push).

Is this code causing a bug?